.pre-commit-config.yaml

```diff
@@ -0,0 +1,65 @@
+repos:
+  - repo: local
+    hooks:
+      # rustup component add rustfmt
+      - id: cargo-fmt
+        name: cargo fmt
+        entry: cargo fmt --all --
+        language: system
+        types: [rust]
+        pass_filenames: false
+      - id: cargo-check
+        name: cargo check
+        entry: cargo check --all-features --all-targets --
+        language: system
+        pass_filenames: false
+        types: [rust]
+      # cargo install --locked cargo-deny
+      - id: cargo-deny
+        name: cargo deny
+        entry: cargo deny --all-features check --
+        language: system
+        pass_filenames: false
+      # rustup component add clippy
+      - id: cargo-clippy
```

Comment:
I don't think we should add `cargo check` and `cargo clippy` in pre-commit hooks because they may take a long time which makes the git commit UX bad.

Reply:
I was able to set these checks to run only for the 'manual' stage which means only when pre-commit is run with `pre-commit run --hook-stage manual --all`.
It's debatable whether another task runner like makefile, [cargo make](https://github.com/sagiegurari/cargo-make) or [just](https://github.com/casey/just) would be easier to use for this, but at least pre-commit is able to serve as a commit hook and also as a way to fairly easily run all the checks at once with a single tool.